### PR TITLE
Collect output scale for quantized op and fused op

### PR DESCRIPTION
**Pull request by juncaipeng**
* Collect output scale for quantized op and fused op in post_training_quantization
* Add api for setting activation_quantize_type and weight_quantize_type
* Add api for saving combined model
* Post_training_quantizaion sets batch_generator to support lod tensor

### CODE REVIEW

**Review by wzzju — APPROVED** [review]
LGTM.

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py [hunk 1]
```diff
@@ -1195,7 +1235,7 @@ def _scale_name(self, var_name):
         return "%s@scale" % (var_name)
 
 
-class ScaleForInferencePass(object):
```
Comment (wzzju) [inline python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:1198]: 改pass名字会影响其他部分的代码吗？

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py [hunk 2]
```diff
@@ -1088,7 +1128,7 @@ def apply(self, graph):
         return graph
 
 
-class ScaleForTrainingPass(object):
+class OutScaleForTrainingPass(object):
```
Comment (wzzju) [inline python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:1131]: 改pass名字会影响其他部分的代码吗？

**Review by wzzju — DISMISSED** [review]
LGTM.

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py (hunk 2 above)
Comment (juncaipeng) [reply python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:1131]: 不会的，我测试过所有分类模型。

**Comment by lidanqing-vv** [comment]
@Sand3r- @wojtuss  Hi, the code adding threshold is in this PR

**Review by wzzju — APPROVED** [review]
LGTM.

**Pull request merged**